L1Trigger/CSCTriggerPrimitives/src/CSCGEMMotherboardME11.cc

```diff
@@ -0,0 +1,976 @@
+{
+  std::vector<CSCCorrelatedLCTDigi> LCTs_final;
+  LCTs_final.clear();
+  for (int bx = 0; bx < MAX_LCT_BINS; bx++)
+    {
+      // get sorted LCTs per subchamber
+      const auto& LCTs1a = sortLCTsByQuality(bx, ME1A);
+      const auto& LCTs1b = sortLCTsByQuality(bx, ME1B);
+
+      // temporary collection with all LCTs in the whole chamber
+      std::vector<CSCCorrelatedLCTDigi> LCTs_tmp;
+      LCTs_tmp.insert(LCTs_tmp.begin(), LCTs1b.begin(), LCTs1b.end());
+      LCTs_tmp.insert(LCTs_tmp.end(), LCTs1a.begin(), LCTs1a.end());
+
+      // sort the selected LCTs
+      LCTs_tmp = sortLCTsByQuality(LCTs_tmp);
+
+      //LCTs reduction per BX
+      if (max_lcts==2)
+        {
+          // loop on all the selected LCTs
+          for (const auto& p: LCTs_tmp){
+            // case when you only want to readout ME1A
+            if (me==ME1A and std::find(LCTs1a.begin(), LCTs1a.end(), p) != LCTs1a.end()){
+              LCTs_final.push_back(p);
+            }
+            // case when you only want to readout ME1B
+            else if(me==ME1B and std::find(LCTs1b.begin(), LCTs1b.end(), p) != LCTs1b.end()){
+              LCTs_final.push_back(p);
+            }
+          }
+        }
+      else {
+        if (!LCTs1a.empty() and !LCTs1b.empty() and me==ME1A)
+          LCTs_final.push_back(*LCTs1a.begin());
+        else if (!LCTs1a.empty() and !LCTs1b.empty() and me==ME1B)
+          LCTs_final.push_back(*LCTs1b.begin());
+        else if (!LCTs1a.empty() and LCTs1b.empty() and me==ME1A)
+          LCTs_final.insert(LCTs_final.end(), LCTs1a.begin(), LCTs1a.end());
+        else if (!LCTs1b.empty() and LCTs1a.empty() and me==ME1B)
+          LCTs_final.insert(LCTs_final.end(), LCTs1b.begin(), LCTs1b.end());
+      }
+    }
+  return LCTs_final;
+}
+
+
+//sort LCTs by GEMDPhi in each BX
+std::vector<CSCCorrelatedLCTDigi> CSCGEMMotherboardME11::sortLCTsByGEMDPhi(int bx, enum CSCPart me) const
+{
+  auto allLCTs(me==ME1A ? allLCTs1a : allLCTs1b);
+  std::vector<CSCCorrelatedLCTDigi> LCTs;
+  std::vector<CSCCorrelatedLCTDigi> tmpV;
+  tmpV.clear();
+  LCTs.clear();
+  for (unsigned int mbx = 0; mbx < match_trig_window_size; mbx++)
+    for (int i=0;i<2;i++)
+      if (allLCTs(bx,mbx,i).isValid())
+        LCTs.push_back(allLCTs(bx,mbx,i));
+
+  std::sort(LCTs.begin(), LCTs.end(), CSCMotherboard::sortByGEMDphi);
+  tmpV = LCTs;
+  if (tmpV.size() > max_lcts) tmpV.erase(tmpV.begin()+max_lcts, tmpV.end());
+  return tmpV;
+}
+
+std::vector<CSCCorrelatedLCTDigi> CSCGEMMotherboardME11::sortLCTsByGEMDPhi(std::vector<CSCCorrelatedLCTDigi> LCTs) const
+{
+  std::vector<CSCCorrelatedLCTDigi> tmpV;
+  tmpV.clear();
+  std::sort(LCTs.begin(), LCTs.end(), CSCMotherboard::sortByGEMDphi);
+  tmpV = LCTs;
+  if (tmpV.size() > max_lcts) tmpV.erase(tmpV.begin()+max_lcts, tmpV.end());
+  return tmpV;
+}
+
+//sort LCTs in whole LCTs BX window
+std::vector<CSCCorrelatedLCTDigi> CSCGEMMotherboardME11::sortLCTsByGEMDPhi(enum CSCPart me) const
```

Comment:
It seems the only difference between this and the corresponding `sortLCTsByQuality` is the sort criterion. Consider make a generic `sortLCTsByX` with the sort criterion passed as a parameter; then you can make the specific versions you want that just call the generic with the desired criterion.

Edit: there's actually an example of this in the next file:
```CSCUpgradeMotherboard::sortLCTs(result, CSCUpgradeMotherboard::sortLCTsByQuality)```

Reply:
I will rework the sortByLCT functions